nav2_rotation_shim_controller/src/nav2_rotation_shim_controller.cpp

```diff
@@ -201,6 +246,17 @@ geometry_msgs::msg::PoseStamped RotationShimController::getSampledPathPt()
             "passing off to primary controller plugin.", forward_sampling_distance_));
 }
 
+geometry_msgs::msg::PoseStamped RotationShimController::getSampledPathGoal()
+{
+  if (current_path_.poses.empty()) {
+    throw nav2_core::InvalidPath("Path is empty - cannot find a goal point");
+  }
+
+  auto goal = current_path_.poses.back();
+  goal.header.stamp = clock_->now();
```

Comment:
Don't restamp - use the path's timestamp `current_path_.header.stamp` so that we remain synchronized

Reply:
When not restamping, I get the following error: 
`[transformPoseInTargetFrame]: Extrapolation Error looking up target frame: Lookup would require extrapolation into the past.  Requested time 11.600000 but the earliest data is at time 18.184000, when looking up transform from frame [map] to frame [odom]`

When I check the function `getSamplePathPt` that also return a point of the path, it is restamping the resulting path point: https://github.com/ros-navigation/navigation2/blob/83be52e470d33508780777cd5a283bfa79b268b4/nav2_rotation_shim_controller/src/nav2_rotation_shim_controller.cpp#L193C27-L193C28

I could not find the exact location where the stamp is set in the `current_path_.header.stamp`, but it seems it is not updated. Am I wrong ?